rust/parquet/src/schema/types.rs

```diff
@@ -419,6 +439,79 @@ impl<'a> PrimitiveTypeBuilder<'a> {
             precision: self.precision,
         })
     }
+
+    #[inline]
+    fn test_decimal_precision_scale(&self) -> Result<()> {
```

Comment:
nit: maybe rename `test` to `check` or something else. It's easy to misread this as a test case.